test_compute_flow.py fails x 4, with "TypeError: string indices must be integers"

### Bug report

I made a simple one-paragraph change to a README, in [this PR](https://github.com/oceanprotocol/ocean.py/pull/1359). 

Yet in remote CI, four tests failed. All sub-tests of `test_compute_flow.py`:

```text
=========================== short test summary info ============================
FAILED tests/integration/ganache/test_compute_flow.py::test_compute_update_trusted_algorithm
FAILED tests/integration/ganache/test_compute_flow.py::test_compute_trusted_publisher
FAILED tests/integration/ganache/test_compute_flow.py::test_compute_just_provider_fees
FAILED tests/readmes/test_readmes.py::test_script_execution[script6] - TypeEr...
```

All four tests had the same failure:
```text
>   return next(env for env in environments if float(env["priceMin"]) == float(0))
E   TypeError: string indices must be integers
```


### Traceback

Full logs are [here](https://github.com/oceanprotocol/ocean.py/actions/runs/4352325867/jobs/7604969865).

Here's one example traceback.

```text
=================================== FAILURES ===================================
____________________ test_compute_update_trusted_algorithm _____________________

publisher_wallet = <LocalAccount '0x02354A1F160A3fd7ac8b02ee91F04104440B28E7'>
publisher_ocean = <ocean_lib.ocean.ocean.Ocean object at 0x7f9ae1bd2d30>
consumer_wallet = <LocalAccount '0x529043886F21D9bc1AE0feDb[75](https://github.com/oceanprotocol/ocean.py/actions/runs/4352325867/jobs/7604969865#step:9:76)1e34265a246e47'>
dataset_with_compute_service_generator = <ocean_lib.assets.ddo.DDO object at 0x7f9ae1cdc6a0>
algorithm = <ocean_lib.assets.ddo.DDO object at 0x7f9ae1bec880>
algorithm_with_different_publisher = <ocean_lib.assets.ddo.DDO object at 0x7f9ae2dc0280>

    @pytest.mark.integration
    def test_compute_update_trusted_algorithm(
        publisher_wallet,
        publisher_ocean,
        consumer_wallet,
        dataset_with_compute_service_generator,
        algorithm,
        algorithm_with_different_publisher,
    ):
        trusted_algo_list = [algorithm.generate_trusted_algorithms()]
        compute_service = get_first_service_by_type(
            dataset_with_compute_service_generator, "compute"
        )
    
        compute_service.update_compute_values(
            trusted_algorithms=trusted_algo_list,
            trusted_algo_publishers=[],
            allow_network_access=True,
            allow_raw_algorithm=False,
        )
    
        updated_dataset = publisher_ocean.assets.update(
            dataset_with_compute_service_generator, {"from": publisher_wallet}
        )
    
        # Expect to pass when trusted algorithm is used
>       run_compute_test(
            ocean_instance=publisher_ocean,
            publisher_wallet=publisher_wallet,
            consumer_wallet=consumer_wallet,
            dataset_and_userdata=AssetAndUserdata(updated_dataset, None),
            algorithm_and_userdata=AssetAndUserdata(algorithm, None),
            scenarios=["with_result"],
        )

tests/integration/ganache/test_compute_flow.py:510: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/integration/ganache/test_compute_flow.py:222: in run_compute_test
    free_c2d_env = ocean_instance.compute.get_free_c2d_environment(
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/ocean/ocean_compute.py:155: in get_free_c2d_environment
    return next(env for env in environments if float(env["priceMin"]) == float(0))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <dict_keyiterator object at 0x7f9ae1d7bd10>

>   return next(env for env in environments if float(env["priceMin"]) == float(0))
E   TypeError: string indices must be integers

ocean_lib/ocean/ocean_compute.py:155: TypeError
````

--- a/.github/workflows/pytest.yml
+++ b/.github/workflows/pytest.yml
@@ -30,7 +30,7 @@ jobs:
       - name: Run Barge
         working-directory: ${{ github.workspace }}/barge
         run: |
-          bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-c2d > start_ocean.log &
+          bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-c2d --with-thegraph --skip-subgraph-deploy > start_ocean.log &
       - name: Install dependencies
         working-directory: ${{ github.workspace }}
         run: |

--- a/.github/workflows/pytest.yml
+++ b/.github/workflows/pytest.yml
@@ -30,7 +30,7 @@ jobs:
       - name: Run Barge
         working-directory: ${{ github.workspace }}/barge
         run: |
-          bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-rbac --with-c2d --with-thegraph > start_ocean.log &
+          bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-c2d > start_ocean.log &
       - name: Install dependencies
         working-directory: ${{ github.workspace }}
         run: |

--- a/tests/flows/test_start_order_fees.py
+++ b/tests/flows/test_start_order_fees.py
@@ -217,7 +217,6 @@ def create_asset_with_order_fee_and_timeout(
     publish_market_order_fees,
     timeout: int,
 ) -> Tuple[DDO, Service, DatatokenBase]:
-
     # Create datatoken with order fee
     datatoken = data_nft.create_datatoken(
         {"from": publisher_wallet},

--- a/tests/integration/ganache/test_compute_flow.py
+++ b/tests/integration/ganache/test_compute_flow.py
@@ -488,15 +488,15 @@ def test_compute_trusted_algorithm(
 
 
 @pytest.mark.integration
-@skip_on(
-    (
-        ocean_lib.exceptions.DataProviderException,
-        requests.exceptions.ConnectionError,
-        TypeError,
-        AssertionError,
-    ),
-    reason="Fix provider issue #606",
-)
+# @skip_on(
+#     (
+#         ocean_lib.exceptions.DataProviderException,
+#         requests.exceptions.ConnectionError,
+#         TypeError,
+#         AssertionError,
+#     ),
+#     reason="Fix provider issue #606",
+# )
 def test_compute_update_trusted_algorithm(
     publisher_wallet,
     publisher_ocean,
@@ -549,7 +549,7 @@ def test_compute_update_trusted_algorithm(
 
 
 @pytest.mark.integration
-@skip_on((TypeError, AssertionError), reason="Fix provider issue #606")
+# @skip_on((TypeError, AssertionError), reason="Fix provider issue #606")
 def test_compute_trusted_publisher(
     publisher_wallet,
     publisher_ocean,
@@ -585,7 +585,7 @@ def test_compute_trusted_publisher(
 
 
 @pytest.mark.integration
-@skip_on((TypeError, AssertionError), reason="Fix provider issue #606")
+# @skip_on((TypeError, AssertionError), reason="Fix provider issue #606")
 def test_compute_just_provider_fees(
     publisher_wallet,
     publisher_ocean,

--- a/tests/integration/ganache/test_compute_flow.py
+++ b/tests/integration/ganache/test_compute_flow.py
@@ -26,7 +26,6 @@ from tests.resources.ddo_helpers import (
     get_registered_asset_with_access_service,
     get_registered_asset_with_compute_service,
 )
-from tests.resources.helper_functions import skip_on
 
 
 @pytest.fixture
@@ -488,15 +487,6 @@ def test_compute_trusted_algorithm(
 
 
 @pytest.mark.integration
-# @skip_on(
-#     (
-#         ocean_lib.exceptions.DataProviderException,
-#         requests.exceptions.ConnectionError,
-#         TypeError,
-#         AssertionError,
-#     ),
-#     reason="Fix provider issue #606",
-# )
 def test_compute_update_trusted_algorithm(
     publisher_wallet,
     publisher_ocean,
@@ -549,7 +539,6 @@ def test_compute_update_trusted_algorithm(
 
 
 @pytest.mark.integration
-# @skip_on((TypeError, AssertionError), reason="Fix provider issue #606")
 def test_compute_trusted_publisher(
     publisher_wallet,
     publisher_ocean,
@@ -585,7 +574,6 @@ def test_compute_trusted_publisher(
 
 
 @pytest.mark.integration
-# @skip_on((TypeError, AssertionError), reason="Fix provider issue #606")
 def test_compute_just_provider_fees(
     publisher_wallet,
     publisher_ocean,

--- a/tests/integration/remote/util.py
+++ b/tests/integration/remote/util.py
@@ -81,7 +81,6 @@ def do_nonocean_tx_and_handle_gotchas(ocean, alice_wallet, bob_wallet):
 
     print("Do a send-Ether tx...")
     try:
-
         priority_fee, _ = get_gas_fees()
         alice_wallet.transfer(
             bob_wallet.address,

--- a/tests/resources/helper_functions.py
+++ b/tests/resources/helper_functions.py
@@ -358,19 +358,3 @@ def get_mock_provider_fees(mock_type, wallet, valid_until=0):
         "validUntil": valid_until,
         "providerData": Web3.toHex(Web3.toBytes(text=provider_data)),
     }
-
-
-def skip_on(exception, reason="Default reason"):
-    """Decorator for skipping test in case of known issue."""
-
-    def decorator_func(f):
-        @wraps(f)
-        def wrapper(*args, **kwargs):
-            try:
-                return f(*args, **kwargs)
-            except exception:
-                pytest.skip(reason)
-
-        return wrapper
-
-    return decorator_func

--- a/tests/resources/mocks/http_client_mock.py
+++ b/tests/resources/mocks/http_client_mock.py
@@ -36,7 +36,7 @@ class HttpClientMockBase(Session):
     def get(cls, *args, **kwargs):
         """Handles the base case of service endpoints."""
         is_get_endpoints_request = False
-        for (_, _, _, fn, _, _) in inspect.getouterframes(inspect.currentframe()):
+        for _, _, _, fn, _, _ in inspect.getouterframes(inspect.currentframe()):
             if fn == "get_service_endpoints":
                 is_get_endpoints_request = True
 

